### ISSUE TITLE
The 1MIN sensor goes offline every now and then

### ISSUE THREAD

**Issue by cooltings** [magico13/ha-emporia-vue#232]
<img width="300" alt="image" src="https://github.com/magico13/ha-emporia-vue/assets/104278596/fa712170-ff5f-4a5b-b2bd-2bd32c3ad73c">
Every now and then live sensor goes offline. I can still see it in emporia app.

**Comment by PYehl73** [comment]
Happens constantly for me as well!

**Comment by sdwhwk** [comment]
One way to try is restarting the integration. It is not a foolproof method, but sometimes it works (although I’ve restarted mine three times already and the Mins are still unavailable).

**Comment by cooltings** [comment]
Yes I do reload the integration and this works but only temporarily as 2days or weeks later it will show unavailable again

**Comment by sdwhwk** [comment]
That’s been pretty much my experience, also.

**Comment by patrickaubertdelaruee** [comment]
Same for me. Reload integration works for a day or two.

**Comment by shawnhaywood** [comment]
Reloaded the integration today and noticed the same "unavailable" message for a 1M entity.  HA says, "This entity is no longer being provided by the emporia_vue integration. If the entity is no longer in use, delete it in settings.".  Sure hope this gets resolved.  Like others, I have this occur frequently, but when I reload it usually fixes itself.

**Comment by sdwhwk** [comment]
It is an annoyance, but the automation shared earlier seems to be a good temporary fix. I forgot how long I have it checked (less that what was stated to use) but I only ran into the issue once since then.

**Comment by sekonicpr** [comment]
> Reloaded the integration today and noticed the same "unavailable" message for a 1M entity.  HA says, "This entity is no longer being provided by the emporia_vue integration. If the entity is no longer in use, delete it in settings.".  Sure hope this gets resolved.  Like others, I have this occur frequently, but when I reload it usually fixes itself. 

Has the same issue, I reloaded the integration, reset HA and router and it still has an unavailable error.

**Comment by shawnhaywood** [comment]
Checked my dashboard on the emporia site, looks like the "second" and "minute" data isn't reporting on there either.

**Comment by metcalfa29** [comment]
> Checked my dashboard on the emporia site, looks like the "second" and "minute" data isn't reporting on there either.

Glad im not the only one experiencing this issue. My seconds and minutes stopped reporting about an hour or so ago as well.

**Comment by sdwhwk** [comment]
I just noticed that too. Checked my devices to see if they were still connected to the WiFi just to make sure and they were.

**Comment by CrazyCoder** [comment]
Looks like backend issue, according to the logs I see:

```
2023-11-19 00:00:12.313 ERROR (MainThread) [custom_components.emporia_vue] Error communicating with Emporia API: 500 Server Error: Internal Server Error for url: https://api.emporiaenergy.com/AppAPI?apiMethod=getDeviceListUsages&deviceGids=...
```

Data has been unavailable for several hours today:

![image](https://github.com/magico13/ha-emporia-vue/assets/122834/4f40f753-a3c4-4dfb-9f1f-870399e6e9b7)

**Comment by kocsenc** [comment]
Same thing as @CrazyCoder , I see no data reported as of around 2PM Nov28 PST. 

![image](https://github.com/magico13/ha-emporia-vue/assets/2752396/735867d0-25a1-4946-bb2d-5aa1f2a17977)

**Comment by CrazyCoder** [comment]
They claim [it's resolved](https://community.emporiaenergy.com/topic/since-11-19-2023-0320-i-have-not-received-data/):

> We apologize for the inconvenience this weekend. We can confirm that service has been restored. We appreciate your feedback and patience.

**Comment by JAudi23** [comment]
Confirmed for now, I had to restart HA and all entities are active again.  I tried just reloading the Integration however the MIN entities that were unavailable did not come back until I restarted HA.

**Comment by CrazyCoder** [comment]
Too late, switched to the local esphome firmware, no more cloud dependency.

![image](https://github.com/magico13/ha-emporia-vue/assets/122834/f7207f7d-e990-41bc-afb7-ab4f624405cb)


![image](https://github.com/magico13/ha-emporia-vue/assets/122834/69634ea7-5ade-49c2-8c0c-0b39c46f6ca7)

**Comment by JAudi23** [comment]
Do you have a link to good instructions to complete the migration over to ESPHome?
I've been tempted, but hesitant.

**Comment by CrazyCoder** [comment]
@JAudi23 See https://community.home-assistant.io/t/emporia-vue-utility-connect/378347/1 and https://github.com/jrouvier/esphome-emporia-vue-utility.
